tests/net/socket/getaddrinfo/prj.conf

```diff
@@ -15,19 +15,48 @@ CONFIG_TEST_RANDOM_GENERATOR=y
 
 # Network address config
 CONFIG_NET_CONFIG_SETTINGS=y
-CONFIG_NET_CONFIG_MY_IPV4_ADDR="192.0.2.1"
-CONFIG_NET_CONFIG_MY_IPV6_ADDR="2001:db8::1"
+CONFIG_NET_CONFIG_MY_IPV4_ADDR="127.0.0.1"
+CONFIG_NET_CONFIG_MY_IPV6_ADDR="::1"
 
 # Enable the DNS resolver
 CONFIG_DNS_RESOLVER=y
 CONFIG_DNS_SERVER_IP_ADDRESSES=y
 
-# Use local dnsmasq server for testing
-CONFIG_DNS_SERVER1="[2001:db8::2]:15353"
-CONFIG_DNS_SERVER2="192.0.2.2:15353"
+# Use local server for testing.
+CONFIG_DNS_SERVER1="[::1]:15353"
+CONFIG_DNS_SERVER2="127.0.0.1:15353"
 
 CONFIG_MAIN_STACK_SIZE=2048
 CONFIG_ZTEST=y
 
 # User mode requirements
 CONFIG_HEAP_MEM_POOL_SIZE=128
+
+# Debugging options
+#CONFIG_NET_LOG=y
+#CONFIG_NET_SOCKETS_LOG_LEVEL_DBG=y
+#CONFIG_NET_CONTEXT_LOG_LEVEL_DBG=y
+#CONFIG_NET_IF_LOG_LEVEL_DBG=y
+#CONFIG_DNS_RESOLVER_LOG_LEVEL_DBG=y
+#CONFIG_NET_CORE_LOG_LEVEL_DBG=y
+#CONFIG_NET_IPV6_LOG_LEVEL_DBG=y
+
+#CONFIG_NET_SHELL=y
+#CONFIG_LOG_IMMEDIATE=y
+
+# We do not need neighbor discovery etc for this test
+CONFIG_NET_IPV6_DAD=n
+CONFIG_NET_IPV6_ND=n
+CONFIG_NET_IPV6_MLD=n
+
+# Disable internal ethernet drivers as the test is self contained
+# and does not need the on board driver to function.
+CONFIG_ETH_NATIVE_POSIX=n
```

Comment:
this doesn't really scale as more drivers are added, is there another way?

Reply:
I do not think this is a big issue as we currently run the test app in qemu_x86 and native_posix. I think we can remove these settings if needed, just need to make sure there is no ill effects. I just added these in order to make sure that only loopback interface will ever be created in the application.

Reply:
ok

Reply:
Removed the disabling of board specific ethernet drivers, the tests work just fine without them.